### ISSUE TITLE
Default filtering rules for k8s

### ISSUE THREAD

**Issue by kartoch** [ovh/terraform-ovh-publiccloud-k8s#44]
Default openstack filtering rules between k8s nodes are weak. From commit 74b0e69bd80ad2839a03919d64a9869e191c9edb  :

```
# auth all ports ; TODO filter only kube ports
resource "openstack_networking_secgroup_rule_v2" "in_traffic_k8s_tcp" {
  count             = "${var.associate_public_ipv4 ? var.count : 0}"
  direction         = "ingress"
  ethertype         = "IPv4"
  protocol          = "tcp"
  remote_ip_prefix  = "${format("%s/32", element(data.template_file.public_ipv4_addrs.*.rendered, count.index))}"
  security_group_id = "${openstack_networking_secgroup_v2.pub.id}"
}

resource "openstack_networking_secgroup_rule_v2" "in_traffic_k8s_udp" {
  count             = "${var.associate_public_ipv4 ? var.count : 0}"
  direction         = "ingress"
  ethertype         = "IPv4"
  protocol          = "udp"
  remote_ip_prefix  = "${format("%s/32", element(data.template_file.public_ipv4_addrs.*.rendered, count.index))}"
  security_group_id = "${openstack_networking_secgroup_v2.pub.id}"
}

resource "openstack_networking_secgroup_rule_v2" "egress-ipv4" {
  count             = "${var.associate_public_ipv4 ? 1 : 0}"
  direction         = "egress"
  ethertype         = "IPv4"
  security_group_id = "${openstack_networking_secgroup_v2.pub.id}"
  remote_ip_prefix  = "0.0.0.0/0"
}
```

For information, the open ports on a master nodes after fresh install:

```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN      2000/kubelet        
tcp        0      0 127.0.0.1:10249         0.0.0.0:*               LISTEN      2421/kube-proxy     
tcp        0      0 127.0.0.1:10251         0.0.0.0:*               LISTEN      2221/kube-scheduler 
tcp        0      0 127.0.0.1:10252         0.0.0.0:*               LISTEN      2250/kube-controlle 
tcp6       0      0 :::4194                 :::*                    LISTEN      2000/kubelet        
tcp6       0      0 :::10250                :::*                    LISTEN      2000/kubelet        
tcp6       0      0 :::9099                 :::*                    LISTEN      2868/calico-felix   
tcp6       0      0 :::6443                 :::*                    LISTEN      2229/kube-apiserver 
tcp6       0      0 :::2379                 :::*                    LISTEN      1452/etcd           
tcp6       0      0 :::2380                 :::*                    LISTEN      1452/etcd           
tcp6       0      0 :::10255                :::*                    LISTEN      2000/kubelet        
tcp6       0      0 :::10256                :::*                    LISTEN      2421/kube-proxy     
tcp6       0      0 :::22                   :::*                    LISTEN      1/systemd           
tcp6       0      0 :::8888                 :::*                    LISTEN      1375/cfssl          
udp        0      0 54.38.71.101:68         0.0.0.0:*                           716/systemd-network 
udp        0      0 0.0.0.0:8472            0.0.0.0:*                           -                   
udp6       0      0 fe80::f816:3eff:fe8:546 :::*                                716/systemd-network 
```

**Comment by kartoch** [comment]
Let's enumerate them

1. traffic between nodes for CNI is based on vxlan, so every node (master or worker) must have an ALLOW on ingress IPv4 UDP port 4789 (warning: it is the calico configuration of this terraform template, could change to GRE or IP-in-IP is calico configuration is modified).

**Comment by kartoch** [comment]
2. Is that a good idea to authorize egress ALL for all node here and not externally (in the provisioning template) ? It will leave the choice to the user to decide if node can communicate by itself (for instance: to be sure it contact only authorized docker registry).

**Comment by kartoch** [comment]
3. The following port are binded on localhost so they must not be open externally:

```
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN      2000/kubelet        
tcp        0      0 127.0.0.1:10249         0.0.0.0:*               LISTEN      2421/kube-proxy     
tcp        0      0 127.0.0.1:10251         0.0.0.0:*               LISTEN      2221/kube-scheduler 
tcp        0      0 127.0.0.1:10252         0.0.0.0:*               LISTEN      2250/kube-controller 
```

**Comment by kartoch** [comment]
4. 4194 is for cAdvisor (inside kubelet) so technically it must be binded to localhost BUT 10250 must be binded externally as it must be accessible for remote log access from `kubectl` operator. Thus I'm not sure if it must be filtered. Need to check how heapster collect the metrics of cAdivsor.

```
tcp6       0      0 :::4194                 :::*                    LISTEN      2000/kubelet        
tcp6       0      0 :::10250                :::*                    LISTEN      2000/kubelet        
```

Based on [Calico DaemonSet Yaml](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/calico-policy-controller/calico-node-daemonset.yaml )

tcp6       0      0 :::9099                 :::*                    LISTEN      2868/calico-felix   
tcp6       0      0 :::6443                 :::*                    LISTEN      2229/kube-apiserver 
tcp6       0      0 :::2379                 :::*                    LISTEN      1452/etcd           
tcp6       0      0 :::2380                 :::*                    LISTEN      1452/etcd           
tcp6       0      0 :::10255                :::*                    LISTEN      2000/kubelet        
tcp6       0      0 :::10256                :::*                    LISTEN      2421/kube-proxy     
tcp6       0      0 :::22                   :::*                    LISTEN      1/systemd           
tcp6       0      0 :::8888                 :::*                    LISTEN      1375/cfssl          
udp        0      0 54.38.71.101:68         0.0.0.0:*                           716/systemd-network 
udp        0      0 0.0.0.0:8472            0.0.0.0:*                           -                   
udp6       0      0 fe80::f816:3eff:fe8:546 :::*                                716/systemd-network

**Comment by kartoch** [comment]
Nice: all ports are described [here](https://github.com/coreos/coreos-kubernetes/blob/master/Documentation/kubernetes-networking.md).

**Comment by kartoch** [comment]
Two remarks:

* Heapster uses external network to access heapster metrics ? That would make sense in case of internal network failure.
* I'm not sure 2379-2380 for the last two lines, I believe
  * master + workers -> etcd: 2379
  * etcd -> etcd: 2380

**Comment by yanndegat** [comment]
fixed by #54
